src/backend/dx11/src/device.rs

```diff
@@ -1208,9 +1328,64 @@ impl hal::Device<Backend> for Device {
 
         let dxgi_format = conv::map_format(image.format).unwrap();
         let decomposed = conv::DecomposedDxgiFormat::from_dxgi_format(dxgi_format);
+        let bpp = format_desc.bits as u32 / 8;
 
         let (view_kind, resource) = match image.kind {
+            image::Kind::D1(width, layers) => {
+                let initial_data = memory.host_visible.as_ref().map(|_p| d3d11::D3D11_SUBRESOURCE_DATA {
+                    pSysMem: unsafe { memory.mapped_ptr.offset(offset as isize) as _ },
+                    SysMemPitch: 0,
+                    SysMemSlicePitch: 0
+                });
+
+                let desc = d3d11::D3D11_TEXTURE1D_DESC {
+                    Width: width,
+                    MipLevels: image.mip_levels as _,
+                    ArraySize: layers as _,
+                    Format: decomposed.typeless,
+                    Usage: usage,
+                    BindFlags: bind,
+                    CPUAccessFlags: cpu,
+                    MiscFlags: 0
+                };
+
+                let mut resource = ptr::null_mut();
+                let hr = unsafe {
+                    self.raw.CreateTexture1D(
+                        &desc,
+                        if let Some(data) = initial_data {
+                            &data
+                        } else {
+                            ptr::null_mut()
+                        },
+                        &mut resource as *mut *mut _ as *mut *mut _
+                    )
+                };
+
+                if !winerror::SUCCEEDED(hr) {
+                    error!("CreateTexture1D failed: 0x{:x}", hr);
+
+                    return Err(device::BindError::WrongMemory);
+                }
+
+                (image::ViewKind::D1Array, unsafe { ComPtr::<d3d11::ID3D11Resource>::from_raw(resource) })
+            },
             image::Kind::D2(width, height, layers, _) => {
+                let mut initial_datas = Vec::new();
+
+                for _layer in 0..layers {
+                    for level in 0..image.mip_levels {
+                        let width = image.kind.extent().at_level(level).width;
+
+                        // TODO: layer offset?
```

Comment:
yeah. If you don't want to apply it, might as well put something like `assert_eq!(layer, 0)` for now